File: IM/InventoryManagement.Core/Models/Inventory.cs
namespace InventoryManagement.Core.Models
{
    public class Inventory : BaseEntity
    {
        public int CompanyId { get; set; }
        public Company Company { get; set; } = default!;


        public int CategoryId { get; set; }
        //public Category Category { get; set; } = default!;



        public int CategorySubId { get; set; }
       // public CategorySub CategorySub { get; set; } = default!;



        public int BrandId { get; set; }
       // public Brand Brand { get; set; } = default!;



        public int ModelId { get; set; }
       // public Model Model { get; set; } = default!;




        public string Name { get; set; } = string.Empty;
        public DateTime InventoryDate { get; set; }
        public DateTime InvoiceDate { get; set; }
        public int Barcode { get; set; }
        public string SerialNumber { get; set; } = string.Empty;
        public int Imei { get; set; }
        public string Mac { get; set; } = string.Empty; 
        public string Status { get; set; } = string.Empty;
        public string Responsible { get; set; } = string.Empty;
    }
}

File: Frontend/src/app/applications/models/inventory/inventory.ts
export class InventoryModel {
  id: number | any;
  name: string | any;
  companyId: number | any;
  categoryId: number | any;
  categorySubId: number | any;
  brandId: number | any;
  modelId: number | any;
  barcode: number | any;
  serialNumber: string | any;
  imei: number | any;
  mac: string | any;
  status: string | any;
  responsible: string | any;
  totalCount: number | any;

  inventoryDate: Date | undefined;
  invoiceDate: Date | undefined;

  createdDate: Date | undefined;
  updatedDate: Date | undefined;
}


export type InventoryListParameters = {
  companyId?: number;
  categoryId?: number;
  categorySubId?: number;
  brandId?: number;
  modelId?: number;
  name?: number;
  barcode?: number;
  serialNumber?: string;
  mac?: string;
  imei?: number;
  responsibleUser?: string;
  status?: string;
}
